### ISSUE TITLE
Lazy getter for static property produce error

### ISSUE THREAD

**Issue by sirian** [microsoft/TypeScript#33493]
**TypeScript Version:**  3.6.3

**Code**

```ts
class A {
    static get callbacks(): any[] {
        delete this.callbacks;
        return this.callbacks = []
    }
}
```

**Expected behavior:**
No error should be shown
![image](https://user-images.githubusercontent.com/897643/65157427-65f42180-da39-11e9-8581-77902a5c1f2c.png)

**Actual behavior:**
error TS2704: The operand of a delete operator cannot be a read-only property.

**Playground Link:** [playground](https://www.typescriptlang.org/play/index.html?target=6#code/MYGwhgzhAECC0G8BQBICAXM6CWxoHMBTdaYMEEAIzGAGsIAKASgC5owA7ATwG0BdREmjCRwgCaEQxQtHQALbBAB0ZCtToQA3ENEiATsQCuejrIXLVVGvWgBeaPx3CAvkldA)

**Comment by sirian** [comment]
Current workaround 
```
class A {
    static get foo(this: any) {
        delete this.foo;
        return this.foo = []
    }
}
```

**Comment by RyanCavanaugh** [comment]
This behavior was requested; see #11730

**Comment by sirian** [comment]
@RyanCavanaugh i think error should be thrown only outside of getter body. Or on explicit `readonly` specifier. But inside getter - not

**Comment by sirian** [comment]
It's common case for lazy getter.
i.e. look at example on MDN https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get#Smart_self-overwriting_lazy_getters

```javascript
get notifier() {
  delete this.notifier;
  return this.notifier = document.getElementById('bookmarked-notification-anchor');
},
```

**Comment by RyanCavanaugh** [comment]
I'm skeptical on "common" given no one else reporting this in the prior 30,000 issues. This is a good use case for `// @ts-ignore`.
